packages/react-components/src/echarts/unboundedZoom/useUnboundedDataZoom.ts

```diff
@@ -87,13 +94,51 @@ const reducer = (state: TickState, action: TickAction): TickState => {
   return state;
 };
 
+const updateGesturesOptions = (
+  gestures: boolean | undefined,
+  dataZoomOption: DataZoomComponentOption,
+  toolboxOption: ToolboxComponentOption
+) => {
+  if (gestures) {
+    // enabled gestures specific option settings
+    dataZoomOption = {
+      ...dataZoomOption,
+      ...DEFAULT_DATA_ZOOM_GESTURES_ENABLED,
+    };
+    toolboxOption = DEFAULT_TOOLBOX_GESTURES_ENABLED;
+  } else {
+    // disabled gestures specific option settings
+    dataZoomOption = {
+      ...dataZoomOption,
+      ...DEFAULT_DATA_ZOOM_GESTURES_DISABLED,
+    };
+    toolboxOption = DEFAULT_TOOLBOX_GESTURES_DISABLED;
+  }
+  return { dataZoomOption, toolboxOption };
+};
+
+const dispatchGesturesEnablement = (
```

Comment:
is this so that things work if the gestures flag changes on the fly?

Reply:
yup, setting config does not deactivate the datazoon selection.